Model01-Firmware.ino

```diff
@@ -279,6 +284,27 @@ void hostPowerManagementEventHandler(kaleidoscope::HostPowerManagement::Event ev
   toggleLedsOnSuspendResume(event);
 }
 
+/** This 'enum' is a list of all the magic combos used by the Model 01's
+ * firmware The names aren't particularly important. What is important is that
+ * each is unique.
+ *
+ * These are the names of your magic combos. They will be used by the
+ * `USE_MAGIC_COMBOS` call below.
+ */
+enum { COMBO_PROTOCOL_TOGGLE };
```

Comment:
COMBO_TOGGLE_NKRO_MODE? 

COMBO_TOGGLE_USB_KEYBOARD_PROTOCOL?

Reply:
`COMBO_TOGGLE_USB_KEYBOARD_PROTOCOL` is more descriptive, but doesn't say what we toggle between, so `COMBO_TOGGLE_NKRO_MODE` is probably better. Simpler to understand too, I think.

Reply:
Sorry for nitpicking...

According to the [Kaleidoscope C++ Coding Style](https://github.com/keyboardio/Kaleidoscope/blob/master/doc/CODING_STYLE.md#enumerator-names) enumerators should preferably be named as constants. Thus, I would suggest 
```cpp
enum { kComboToggleNKROMode };
```

But as `kComboToggleNKROMode` is actually a simple constant rather than an enumerator (only a single value),
```cpp
constexpr uint8_t kComboToggleNGROMode = 0;
```
might be even better.

Reply:
> According to the Kaleidoscope C++ Coding Style enumerators should preferably be named as constants

That part of the coding style has not been updated yet O:)

We don't actually use that style, and will update the coding style guide to match our usage instead.

> But as kComboToggleNKROMode is actually a simple constant rather than an enumerator (only a single value),

The reason I opted for an enum is because it's a bit more future-proof. If we ever wanted to add another magic combo to the default firmware, adding to the enum is easier.

Reply:
> and will update the coding style guide to match our usage instead

Isn't that a bit against the basic idea of a coding style guide?

Reply:
It would be, if the guide weren't a big work in progress, see keyboardio/Kaleidoscope#134.

Reply:
Thanks for pointing me to the WIP issue.

Reply:
What's the rationale behind the leading 'k'? As a non-C++ native, I find it confusing and opaque.

On enum vs single definition: Part of the value of the default sketch is giving the user a framework for adding their own behaviors. Because of that, I think the enum makes more sense.

Stepping back a tiny little bit - why do we even need to name the magic combos like that?

Reply:
Like `kSomething`? Or giving it a name? We don't *need to* give it a name.

```c++
USE_MAGIC_COMBOS({.action = toggleKeyboardProtocol,
                                             // Left Fn + Esc + Shift
                                             .keys = { R3C6, R2C6, R3C7 }
                                          });
```

The above works just as well. The name becomes important*ish* when you start using the same action for multiple combos. We don't do that here, and I don't see it happening in the near - or far - future either.

I can easily drop the enum in this case.

Reply:
The leading 'k' was taken from the current Kaleidoscope coding style guide. Personally, I wouldn't add a 'k' to constants but I am a fan of style conventions.